File: 二叉树/路径总和.go
package 二叉树

func hasPathSum(node *TreeNode, number int) bool { //从该节点到叶子节点，是否存在路径和为number
	if node == nil {
		return false
	}

	if node.Left == nil && node.Right == nil {
		return node.Val == number
	}

	return hasPathSum(node.Left, number-node.Val) || hasPathSum(node.Right, number-node.Val)
}

File: 二叉树/路径总和.java
package 二叉树;

//是否存在一个从跟节点到叶子节点的路径，使得路径和为TargetSum
public class 路径总和 {
    public boolean hasPathSum(TreeNode root, int targetSum) {
        if (root == null) return false;
        if (root.left == null && root.right == null) return root.val == targetSum;
        return hasPathSum(root.left, targetSum - root.val) || hasPathSum(root.right, targetSum - root.val);
    }
}
